### PR TITLE
Support 'oc explain' for Origin types

### PR DESCRIPTION
**Pull request by smarterclayton**
Will now explain Origin types, and @stevekuznetsov's changes will enhance this as well.

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by kargakis** [comment]
Isn't oc explain superseding oc types?

**Comment by smarterclayton** [comment]
Yes, except explain needs to also like our types, which is dependent
on the same underlying problem as the error on edit from RESTMapper.

**Comment by smarterclayton** [comment]
@liggitt for your review

**Comment by smarterclayton** [comment]
Fixes #7125

File: pkg/cmd/util/clientcmd/factory.go [hunk 1]
```diff
@@ -452,6 +468,25 @@ func (f *Factory) Clients() (*client.Client, *kclient.Client, error) {
...
 }
 
+// OriginSwaggerSchema returns a swagger API doc for an Origin schema under the /oapi prefix.
+func (f *Factory) OriginSwaggerSchema(client *kclient.RESTClient, version unversioned.GroupVersion) (*swagger.ApiDeclaration, error) {
+	if version.IsEmpty() {
+		return nil, fmt.Errorf("groupVersion cannot be empty")
+	}
+	path := "/swaggerapi/oapi/" + version.Version
+
+	body, err := client.Get().AbsPath(path).Do().Raw()
```
Comment (liggitt) [inline pkg/cmd/util/clientcmd/factory.go]: `.AbsPath("/").Suffix("swaggerapi","oapi",version.Version)`

File: pkg/cmd/util/clientcmd/factory.go [hunk 2]
```diff
@@ -452,6 +468,25 @@ func (f *Factory) Clients() (*client.Client, *kclient.Client, error) {
 	return osClient, kClient, nil
 }
 
+// OriginSwaggerSchema returns a swagger API doc for an Origin schema under the /oapi prefix.
+func (f *Factory) OriginSwaggerSchema(client *kclient.RESTClient, version unversioned.GroupVersion) (*swagger.ApiDeclaration, error) {
+	if version.IsEmpty() {
+		return nil, fmt.Errorf("groupVersion cannot be empty")
+	}
+	path := "/swaggerapi/oapi/" + version.Version
```
Comment (liggitt) [inline pkg/cmd/util/clientcmd/factory.go]: sort of surprised our policy allows this... is it installed outside the protected handler?

Comment (liggitt) [inline pkg/cmd/util/clientcmd/factory.go]: won't work with v1beta3... do you care?

Comment (smarterclayton) [inline pkg/cmd/util/clientcmd/factory.go]: No

**Comment by smarterclayton** [comment]
[merge] presupposing lgtm

File: pkg/cmd/util/clientcmd/factory.go [hunk 3]
```diff
@@ -452,6 +468,23 @@ func (f *Factory) Clients() (*client.Client, *kclient.Client, error) {
 	return osClient, kClient, nil
 }
 
+// OriginSwaggerSchema returns a swagger API doc for an Origin schema under the /oapi prefix.
+func (f *Factory) OriginSwaggerSchema(client *kclient.RESTClient, version unversioned.GroupVersion) (*swagger.ApiDeclaration, error) {
+	if version.IsEmpty() {
+		return nil, fmt.Errorf("groupVersion cannot be empty")
+	}
+	body, err := client.Get().AbsPath("/").Suffix("swaggerapi", "oapi", version.Version).Do().Raw()
```
Comment (liggitt) [inline pkg/cmd/util/clientcmd/factory.go]: Can this get invoked more than once per process? The swagger doc is huge

**Comment by liggitt** [comment]
Test in test-cmd?

**Comment by liggitt** [comment]
Lgtm otherwise

**Comment by smarterclayton** [comment]
Will add a test in follow up

File: pkg/cmd/util/clientcmd/factory.go (hunk 3 above)
Comment (smarterclayton) [inline pkg/cmd/util/clientcmd/factory.go]: Explain calls it only once

Comment (liggitt) [inline pkg/cmd/util/clientcmd/factory.go]: ok

**Pull request merged**